### PR TITLE
Remove progress model, refactor reports section

### PR DESCRIPTION
**Pull request by afiore**
- refactor reports query to use optimal SQL queries
- refactor report presentation logic into presenter classes
- remove Progress model and progress-changer class

### CODE REVIEW

File: app/controllers/reports_controller.rb [hunk 1]
```diff
@@ -1,68 +1,33 @@
...
+  PER_PAGE = 15
 
   def ministers_by_progress
     update_page_title('Minister report')
-    @ministers = Minister.active
-    @progresses = Progress.where(name: Progress.in_progress_questions)
-
-    @counters = Pq.ministers_by_progress(@ministers, @progresses)
+    report_data = Pq.in_progress_by_minister
+    @report     = Report.ministers(report_data, Minister.active)
```
Comment (foxjerem) [inline app/controllers/reports_controller.rb]: namespace the report as a presenter e.g here Presenters::Report

File: app/models/pq.rb [hunk 2]
```diff
@@ -58,12 +58,21 @@ def rejected
...
 
-  before_update :set_pod_waiting, :process_date_for_answer
+  before_update :set_pod_waiting, :process_date_for_answer, :set_state_weight
 
   def set_pod_waiting
     self.pod_waiting = draft_answer_received if draft_answer_received_changed?
   end
 
+  def set_state_weight
+    self.state_weight=PQState.state_weight(state)
```
Comment (foxjerem) [inline app/models/pq.rb]: spaces

File: app/models/pq_scopes.rb [hunk 3]
```diff
@@ -10,97 +10,148 @@ def not_seen_by_finance
...
+        acc.merge(r.state => h ) { |_, old_v, new_v| old_v.merge(new_v) }
+      }
+  end
+
+  # Returns a hash with the PQ state as key and count by press_desk_id as value.
+  #
+  # It will filter out PQs that are below the PQState::DRAFT_PENDING state,
+  # and press desks that are not active.
+  #
+  def accepted_by_press_desk
```
Comment (foxjerem) [inline app/models/pq_scopes.rb]: should we return a list of Pqs here and then reduce within the controller? Else rename function as implies PQ list returned

File: lib/pq_state/progress_changer.rb [hunk 4]
```diff
@@ -1,64 +1,63 @@
...
-    :minister_cleared,
-    :answered
+    UNASSIGNED,
+    REJECTED,
+    NO_RESPONSE,
+    DRAFT_PENDING,
+    WITH_POD,
+    POD_CLEARED,
+    WITH_MINISTER,
+    MINISTERIAL_QUERY,
```
Comment (foxjerem) [inline lib/pq_state/progress_changer.rb]: duplicate of PQState::ALL

File: lib/pq_state/progress_changer.rb [hunk 5]
```diff
@@ -1,64 +1,63 @@
...
+    MINISTER_CLEARED,
+    ANSWERED
   ]
 
   FINAL_STATES = [
-    :answered,
-    :transferred_out
+    ANSWERED,
+    TRANSFERRED_OUT
```
Comment (foxjerem) [inline lib/pq_state/progress_changer.rb]: Duplicate of PQState::CLOSED

File: app/models/pq_counts.rb [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+      .select("pqs.state, ao.press_desk_id, count(distinct pqs.id)")
+      .where("state != ?", PQState::UNASSIGNED)
+      .where("aopq.response = 'accepted' AND pd.deleted = false")
+      .group('state, ao.press_desk_id')
+      .reduce({}) { |acc, r|
+        h = { r.press_desk_id => r.count }
+        acc.merge(r.state => h ) { |_, old_v, new_v| old_v.merge(new_v) }
+      }
+  end
+end
```
Comment (afiore) [inline app/models/pq_counts.rb]: there's another method which you might want to add here: `Pq.counts_by_state`

**Pull request merged**